[2024-11-08T13:07:58.002+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-08T13:07:58.033+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: api_test_v1.spark-submit manual__2024-11-08T10:57:45.506420+00:00 [queued]>
[2024-11-08T13:07:58.043+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: api_test_v1.spark-submit manual__2024-11-08T10:57:45.506420+00:00 [queued]>
[2024-11-08T13:07:58.044+0000] {taskinstance.py:2865} INFO - Starting attempt 2 of 6
[2024-11-08T13:07:58.060+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): spark-submit> on 2024-11-08 10:57:45.506420+00:00
[2024-11-08T13:07:58.067+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'api_test_v1', 'spark-submit', 'manual__2024-11-08T10:57:45.506420+00:00', '--job-id', '27', '--raw', '--subdir', 'DAGS_FOLDER/Weather_etl_dag.py', '--cfg-path', '/tmp/tmpk9va4a80']
[2024-11-08T13:07:58.071+0000] {standard_task_runner.py:105} INFO - Job 27: Subtask spark-submit
[2024-11-08T13:07:58.072+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=603) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-11-08T13:07:58.074+0000] {standard_task_runner.py:72} INFO - Started process 606 to run task
[2024-11-08T13:07:58.122+0000] {task_command.py:467} INFO - Running <TaskInstance: api_test_v1.spark-submit manual__2024-11-08T10:57:45.506420+00:00 [running]> on host 39fedacfa7f1
[2024-11-08T13:07:58.192+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Simo' AIRFLOW_CTX_DAG_ID='api_test_v1' AIRFLOW_CTX_TASK_ID='spark-submit' AIRFLOW_CTX_EXECUTION_DATE='2024-11-08T10:57:45.506420+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-08T10:57:45.506420+00:00'
[2024-11-08T13:07:58.194+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-08T13:07:58.216+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-11-08T13:07:58.218+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /dags/pysparksubmitoperator.py
[2024-11-08T13:07:58.255+0000] {spark_submit.py:634} INFO - /home/***/.local/lib/python3.12/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-11-08T13:07:58.256+0000] {spark_submit.py:634} INFO - JAVA_HOME is not set
[2024-11-08T13:07:58.264+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /dags/pysparksubmitoperator.py. Error code is: 1.
[2024-11-08T13:07:58.271+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=api_test_v1, task_id=spark-submit, run_id=manual__2024-11-08T10:57:45.506420+00:00, execution_date=20241108T105745, start_date=20241108T130758, end_date=20241108T130758
[2024-11-08T13:07:58.285+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-08T13:07:58.287+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 27 for task spark-submit (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /dags/pysparksubmitoperator.py. Error code is: 1.; 606)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --conf spark.executor.memory=2g --conf spark.executor.cores=1 --name arrow-spark --deploy-mode client /dags/pysparksubmitoperator.py. Error code is: 1.
[2024-11-08T13:07:58.331+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-08T13:07:58.346+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-08T13:07:58.350+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
